app/models/authenticator/base.rb

```diff
@@ -135,14 +135,15 @@ def authorize(taskid, username, *args)
             return nil
           end
 
-          matching_groups = match_groups(groups_for(identity))
+          incoming_groups = groups_for(identity)
+          matching_groups = match_groups(incoming_groups)
           userid, user = find_or_initialize_user(identity, username)
           update_user_attributes(user, userid, identity)
           audit_new_user(audit, user) if user.new_record?
           user.miq_groups = matching_groups
 
           if matching_groups.empty?
-            msg = "Authentication failed for userid #{user.userid}, unable to match user's group membership to an EVM role"
+            msg = "Authentication failed for userid #{user.userid}, unable to match user's group membership to an EVM role. The incoming groups are: #{incoming_groups.join(", ")}"
```

Comment:
I wasn't sure if this should be one line or two separate lines.  I did it this way to get it into both the regular log and the audit log, but I'm not sure it needs to be in the audit log.  @jrafanie What do you think?

Reply:
I like this idea.  We hit this often with podified.  Do we need to have logging on the `match_groups` sided?  In other words, we are logging the `incoming_groups` here but do we also need to log the groups defined in miq?

Reply:
I think this makes sense to keep the log line here so it's in the AuditEvent table along with the audit logging.

Reply:
>  In other words, we are logging the incoming_groups here but do we also need to log the groups defined in miq?

I don't think so.  It's convenient maybe to have it in one spot, but it's also separately checkable.  That is, a user could go to the UI Access Control page and see the list of groups.  The incoming groups on the other hand cannot be seen anywhere, so it's a complete mystery.  I was concerned about printing the internal group names, but I don't know why.  I'm on the fence here.